src/blob/core.rs

```diff
@@ -282,15 +289,17 @@ where
         }))
     }
 
-    pub(crate) async fn mark_all_as_deleted(&mut self, key: &K) -> Result<Option<u64>> {
-        if self.index.get_any(key).await?.is_some() {
-            let on_disk = self.index.on_disk();
+    pub(crate) async fn mark_all_as_deleted(blob: &ASRwLock<Self>, key: &K) -> Result<Option<u64>> {
```

Comment:
The lock in the new implementation is acquired multiple times, which can lead to performance degradation. The original implementation was better. The only possible improvement here can be achieved by acquiring `upgradable_read` lock at first, then check key presence (`write.index.get_any(key).await?.is_some()`) and then upgrade to write lock if it is presented (any further lock manipulations should be avoided)

Reply:
I just saw a removal implementation in `src/storage/core.rs`. I think, it is better just to revert to the original form of this function